risc0/zkvm/benches/fib.rs

```diff
@@ -15,160 +15,202 @@
+
+fn total_composite(c: &mut Criterion) {
+    let mut group = c.benchmark_group("fib");
+    group.sample_size(10);
+
+    let opts = ProverOpts::fast();
+    let prover = get_prover_server(&opts).unwrap();
+    let ctx = VerifierContext::default();
+
+    for iterations in [100, 10_000, 100_000] {
+        let session = setup_exec(iterations).run().unwrap();
+        let id = BenchmarkId::from_parameter(format!("{iterations}/composite"));
+        group.throughput(Throughput::Elements(session.total_cycles));
+        group.bench_function(id, |b| {
+            b.iter_batched_ref(
+                || setup_exec(iterations),
+                |exec| {
+                    black_box({
+                        let session = exec.run().unwrap();
+                        prover.prove_session(&ctx, &session).unwrap()
+                    })
+                },
+                BatchSize::SmallInput,
+            )
+        });
     }
+}
+
+fn total_succinct(c: &mut Criterion) {
+    let mut group = c.benchmark_group("fib");
+    group.sample_size(10);
+
+    let opts = ProverOpts::default();
+    let prover = get_prover_server(&opts).unwrap();
+    let ctx = VerifierContext::default();
+
+    for iterations in [100, 10_000, 100_000] {
+        let session = setup_exec(iterations).run().unwrap();
+        let id = BenchmarkId::from_parameter(format!("{iterations}/succinct"));
+        group.throughput(Throughput::Elements(session.total_cycles));
+        group.bench_function(id, |b| {
+            b.iter_batched_ref(
+                || setup_exec(iterations),
+                |exec| {
+                    black_box({
+                        let session = exec.run().unwrap();
+                        let receipt = prover.prove_session(&ctx, &session).unwrap();
+                        let composite_receipt = receipt.inner.composite().unwrap();
+                        prover.compress(composite_receipt).unwrap();
+                    })
+                },
+                BatchSize::SmallInput,
+            )
+        });
+    }
+}
+
+fn lift(c: &mut Criterion) {
+    let mut group = c.benchmark_group("fib");
+    group.sample_size(10);
+
+    let opts = ProverOpts::default();
+    let prover = get_prover_server(&opts).unwrap();
+    let ctx = VerifierContext::default();
+
+    let id = BenchmarkId::from_parameter(format!("lift"));
+    group.throughput(Throughput::Elements(1 << RECURSION_PO2));
+    group.bench_function(id, |b| {
+        b.iter_batched_ref(
+            || {
+                let mut exec = setup_exec(100);
+                let session = exec.run().unwrap();
+                let segment = session.segments[0].resolve().unwrap();
+                prover.prove_segment(&ctx, &segment).unwrap()
+            },
+            |receipt| black_box(prover.lift(&receipt).unwrap()),
+            BatchSize::SmallInput,
+        );
+    });
+}
+
+fn join(c: &mut Criterion) {
+    let mut group = c.benchmark_group("fib");
+    group.sample_size(10);
```

Comment:
Actually the problem with `join` was that it was trying to combine only a single receipt, which basically meant it was doing nothing, which is why it looked like it was running so fast. So, we actually need to turn down the sample size so that it will not take forever to run.